interfaces/cython/cantera/examples/thermo/equivalenceRatio.py

```diff
@@ -2,52 +2,116 @@
+# Set the mixture composition according to the stoichiometric mixture
+# (equivalence ratio phi = 1). The fuel composition in this example
+# is set to 100 mol-% CH4 and the oxidizer to 21 mol-% O2 and
+# 79 mol-% N2.
+gas.set_equivalence_ratio(phi=1.0, fuel="CH4:1", oxidizer="O2:0.21,N2:0.79")
 
-gas.TP = 300, ct.one_atm
+# The following is a short hand notation and equivalent to the call above
+gas.set_equivalence_ratio(1.0, "CH4", "O2:0.21,N2:0.79")
 
-# set the mixture composition according to the stoichiometric mixture
-# (equivalence ratio = 1)
-gas.set_equivalence_ratio(1, fuel, oxidizer)
+# By default, the composition of fuel and oxidizer are interpreted as mole
+# fractions. If the compositions are given in mass fractions, an
+# additional argument can be provided. Here, the fuel is 100 mass-% CH4
+# and the oxidizer is 23.3 mass-% O2 and 76.7 mass-% N2.
+gas.set_equivalence_ratio(1.0, "CH4", "O2:0.233,N2:0.767", basis='mass')
 
 # This function can be used to compute the equivalence ratio for any mixture.
-# An optional argument "basis" indicates if fuel and oxidizer compositions are
-# provided in terms of mass or mole fractions. Default is mole fractions.
-# If fuel and oxidizer are given in mass fractions, use basis='mass'
-phi = gas.equivalence_ratio(fuel, oxidizer)
-print("phi = {:1.3f}".format(phi))
-
-# The equivalence ratio can also be computed from the elemental composition
-# assuming that there is no oxygen in the fuel and no C,H and S elements
-# in the oxidizer so that the composition of fuel and oxidizer can be omitted
+# The first two arguments specify the compositions of the fuel and oxidizer.
+# An optional third argument "basis" indicates if fuel and oxidizer compositions
+# are provided in terms of mass or mole fractions. Default is mole fractions.
+# Note that for all functions shown here, the compositions are normalized
+# internally so the species fractions do not have to sum to unity
+phi = gas.equivalence_ratio(fuel="CH4", oxidizer="O2:233,N2:767", basis='mass')
+print(f"phi = {phi:1.3f}")
+
+# If the compositions of fuel and oxidizer are unknown, the function can
+# be called without arguments. This assumes that all C, H and S atoms come from
+# the fuel and all O atoms from the oxidizer. In this example, the fuel was set
+# to be pre CH4 and the oxidizer O2:0.233,N2:0.767 so that the assumption is true
+# and the same equivalence ratio as above is computed
 phi = gas.equivalence_ratio()
+print(f"phi = {phi:1.3f}")
 
-# In this example, the result is the same as above
-print("phi = {:1.3f}".format(phi))
+# Instead of working with equivalence ratio, mixture fraction can be used.
+# The mixture fraction is always kg fuel / (kg fuel + kg oxidizer), independent
+# of the basis argument. For example, the mixture fraction Z can be computed as
+# follows. Again, the compositions by default are interpreted as mole fractions
+Z = gas.mixture_fraction(fuel="CH4", oxidizer="O2:0.21,N2:0.79")
+print(f"Z = {Z:1.3f}")
```

Comment:
```suggestion
print(f"Z = {Z:1.3f}")

```
(missing line break)